tests/test-dirs/expand_node/ppx-tests.t/run.t

```diff
@@ -0,0 +1,579 @@
+  > EOF
+
+  $ dune build
+
+  $ $MERLIN single expand-ppx -position 1:22 -filename ./apy.mli < ./apy.mli
+  {
+    "class": "return",
+    "value": {
+      "code": "include sig [@@@ocaml.warning \"-32\"] type pppp_renamed = .. end[@@ocaml.doc
+                                                                   \"@inline\"]
+  [@@merlin.hide ]",
+      "deriver": {
+        "start": {
+          "line": 1,
+          "col": 15
+        },
+        "end": {
+          "line": 1,
+          "col": 34
+        }
+      }
+    },
+    "notifications": []
+  }
+
+Exception in structure
+  $ cat > apr.ml << EOF
+  > exception Foo of string [@@deriving rename]
+  > EOF
+
+  $ dune build
+
+  $ $MERLIN single expand-ppx -position 1:30 -filename ./apr.ml < ./apr.ml
+  {
+    "class": "return",
+    "value": {
+      "code": "include struct exception Foo_renamed of string  end[@@ocaml.doc \"@inline\"]
+  [@@merlin.hide ]",
+      "deriver": {
+        "start": {
+          "line": 1,
+          "col": 24
+        },
+        "end": {
+          "line": 1,
+          "col": 43
+        }
+      }
+    },
+    "notifications": []
+  }
+
+Exception in signature
+  $ cat > apr.mli << EOF
+  > exception Foo of string [@@deriving rename]
+  > EOF
+
+  $ dune build
+
+  $ $MERLIN single expand-ppx -position 1:30 -filename ./apr.mli < ./apr.mli
+  {
+    "class": "return",
+    "value": {
+      "code": "include sig [@@@ocaml.warning \"-32\"] exception Foo_renamed of string  end
+  [@@ocaml.doc \"@inline\"][@@merlin.hide ]",
+      "deriver": {
+        "start": {
+          "line": 1,
+          "col": 24
+        },
+        "end": {
+          "line": 1,
+          "col": 43
+        }
+      }
+    },
+    "notifications": []
+  }
+
+Module type declaration in structure
+  $ cat > apc.ml << EOF
+  > module type Stack = sig 
+  >   type t [@@deriving rename]
+  >   type stack 
+  >   val empty : stack
+  >   val is_empty : stack -> bool
+  >   val push : t -> stack -> stack 
+  >   val pop : stack -> stack
+  >   val peek : stack -> t
+  > end [@@deriving rename]
+  > EOF
+
+  $ dune build
+
+a cursor here should only output the derived t 
+  $ $MERLIN  single expand-ppx -position 2:14 -filename ./apc.ml < ./apc.ml
+  {
+    "class": "return",
+    "value": {
+      "code": "include sig [@@@ocaml.warning \"-32\"] type t_renamed end[@@ocaml.doc
+                                                           \"@inline\"][@@merlin.hide
+                                                                      ]
+  include sig [@@@ocaml.warning \"-32\"] type t_renamed end[@@ocaml.doc
+                                                           \"@inline\"][@@merlin.hide
+                                                                      ]
+  include sig [@@@ocaml.warning \"-32\"] type t_renamed end[@@ocaml.doc
+                                                           \"@inline\"][@@merlin.hide
+                                                                      ]",
```

Comment:
Are the duplications in this test and the following one expected ?

Reply:
Yes, but it's unrelated to Merlin or the functionality itself. The problem is with the example ppx I have written. Somewhere in the implementation for module type declarations, it takes the original type, and while processing, processes the ppx again for that type.

Reply:
Right, that seems like a non-trivial fix...

Maybe we should just rely on additional dependencies for testing, would `ppx_deriving` allows you to simplify the tests and use "real" ppxes ?

Reply:
yes we could, but it may be difficult to find a ppx that acts on nodes like `module_type_declarations` or `exceptions`

Reply:
Right, let's keep it like that then. It would be great if you could add a comment in the test explaining why there are duplicated results. Then I will do a last review before merging :-)

Reply:
I just added the comment